.github/workflows/build_and_test.yml

```diff
@@ -134,6 +134,13 @@ jobs:
         cd pytket/tests
         pip install -r requirements.txt
         pytest --ignore=simulator/ --doctest-modules
+    - name: Upload pytket coverage artefact
+      if: github.event_name == 'pull_request'
```

Comment:
On a related note... the pytket tests are run with python 3.9 for a PR, but python 3.8 for a push to develop... do you think this will cause some fluctuations in the coverage report? Do we need to maintain separate reports for the different python versions (including 3.10 used with the scheduled runs)?

Reply:
Good question, but I'd be quite surprised if it made any difference. I don't think it's worth maintaining three different coverage reports.

Reply:
Agreed. I've fixed the report generation and artifact uploads for the push case.